frontend/src/scenes/cohorts/cohortLogic.tsx

```diff
@@ -104,27 +104,28 @@ export const cohortLogic = kea<cohortLogicType>({
     listeners: ({ actions, values, key }) => ({
         saveCohort: async ({ cohortParams, filterParams }, breakpoint) => {
             let cohort = { ...values.cohort, ...cohortParams } as CohortType
+            const isNewCohort = typeof cohort.id !== 'number'
             const cohortFormData = new FormData()
 
             for (const [itemKey, value] of Object.entries(cohort as CohortType)) {
                 if (itemKey === 'groups') {
                     if (cohort.is_static) {
-                        if (!cohort.csv && cohort.id === 'new') {
+                        if (!cohort.csv && isNewCohort) {
                             actions.setSubmitted(true)
-                            return
+                            continue
                         }
                     } else {
                         for (const _group of value) {
                             if (_group.matchType === PROPERTY_MATCH_TYPE && !_group.properties?.length) {
                                 // Match group should have at least one property
                                 actions.setSubmitted(true)
-                                return
+                                continue
```

Comment:
Tangential, but we should probably call `actions.setSubmitted(true)` at the start of the listener regardless, if it's valid, error messages should not show up anyways, and it follows the natural logic of this.